libs/application/templates/new-primary-school/src/forms/NewPrimarySchoolForm/primarySchoolSection/currentNurserySubSection.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  buildAsyncSelectField,
+  buildMultiField,
+  buildSubSection,
+  coreErrorMessages,
+} from '@island.is/application/core'
+import { friggSchoolsByMunicipalityQuery } from '../../../graphql/queries'
+import { ApplicationType } from '../../../lib/constants'
+import { newPrimarySchoolMessages } from '../../../lib/messages'
+import { getApplicationAnswers } from '../../../lib/newPrimarySchoolUtils'
+import { FriggSchoolsByMunicipalityQuery } from '../../../types/schema'
+
+export const currentNurserySubSection = buildSubSection({
+  id: 'currentNurserySubSection',
+  title: newPrimarySchoolMessages.primarySchool.currentNurserySubSectionTitle,
+  condition: (answers) => {
+    // Only display section if application type is "Enrollment in primary school"
+    const { applicationType } = getApplicationAnswers(answers)
+    return applicationType === ApplicationType.ENROLLMENT_IN_PRIMARY_SCHOOL
+  },
+  children: [
+    buildMultiField({
+      id: 'currentNursery',
+      title:
+        newPrimarySchoolMessages.primarySchool.currentNurserySubSectionTitle,
+      children: [
+        buildAsyncSelectField({
+          id: 'currentNursery.municipality',
+          title: newPrimarySchoolMessages.shared.municipality,
+          placeholder: newPrimarySchoolMessages.shared.municipalityPlaceholder,
+          loadingError: coreErrorMessages.failedDataProvider,
+          dataTestId: 'current-nursery-municipality',
+          loadOptions: async ({ apolloClient }) => {
+            const { data } =
+              await apolloClient.query<FriggSchoolsByMunicipalityQuery>({
+                query: friggSchoolsByMunicipalityQuery,
+              })
+
+            return (
+              data?.friggSchoolsByMunicipality?.map((municipality) => ({
+                value: municipality.name,
+                label: municipality.name,
```

Comment:
Can use object destructuring
```suggestion
              data?.friggSchoolsByMunicipality?.map(({ name }) => ({
                value: name,
                label: name,
```

Reply:
Fixed